src/App.js

```diff
@@ -14,11 +14,87 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    function goodsSelector(goods) {
+      switch (goods.length) {
+        case 0:
+          return 'No goods selected';
+        case 1:
+          return `${goods[0]} is selected`;
+        default:
+          return `${goods.slice(0, goods.length - 1).join(', ')}`
+          + ` and ${goods[goods.length - 1]} are selected`;
+      }
+    }
+
+    const removeGood = (goods, good) => {
+      this.setState({ selectedGoods: goods
+        .filter(product => product !== good) });
+    };
+
+    const addGood = (goods, good) => {
+      this.setState({ selectedGoods: [...goods, good] });
+    };
+
+    const clearAllGoods = () => {
+      this.setState({ selectedGoods: [] });
+    };
+
+    return (
+      <div className="App">
+        <div className="App__wrapper">
+          <h1 className="App__header">
+            {goodsSelector(selectedGoods)}
+          </h1>
+          <button
+            type="button"
+            className={
+              selectedGoods.length === 0
+                ? 'App__clear-button--disable'
+                : 'App__clear-button'
+            }
+            onClick={() => {
+              clearAllGoods();
+            }}
+          >
+            ✗
+          </button>
+        </div>
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              className={
+                selectedGoods.includes(good)
+                  ? 'App__good App__good--active'
+                  : 'App__good'
+              }
+              key={good}
+            >
+              <span>{good}</span>
+              <button
+                className="App__button"
+                type="button"
+                onClick={() => (
+                  selectedGoods.includes(good)
+                    ? removeGood(selectedGoods, good)
+                    : addGood(selectedGoods, good)
+                )}
+              >
+                {selectedGoods.includes(good) ? 'Remove' : 'Add'}
+              </button>
```

Comment:
let's render 2 different buttons depending on the condition

Reply:
Fix all according to your review but I already have 1 button which changes text depending on the condition. So instead I add 2 different buttons for adding and removing goods.